### ISSUE TITLE
Implement Field-Level Scoring in Benchmark Harness

### ISSUE THREAD

**Issue by Kevin-TCUS** [The-Chamber-of-Us/libelle#82]
# Implement Field-Level Scoring in Benchmark Harness

## 🔗 Relationship to prior work
This is a logic enhancement to the benchmark harness currently being finalized in **#77**.

## 🎯 The Upgrade
Right now the thinking is: **“Does the parsed result match the sample resume?”** We need to make it more structured to gain better insight into where the parser is succeeding and where it is failing.

---

## 🛠️ What you need to do
Move from a "whole resume score" to individual **field-level scoring functions**.

### 1) Implement Functional Scorers
Add specific functions for each field. Each returns a float between **0.0 and 1.0**:
* `score_name()`
* `score_email()`
* `score_location()`
* `score_skills()`

### 2) Use Fuzzy Logic (Example)
Instead of a binary True/False, use nuanced returns to handle "near misses":

```python
def score_email(pred, truth):
    if pred == truth:
        return 1.0
    elif pred and truth and pred.lower() == truth.lower():
        return 0.9  # Case mismatch
    elif pred:
        return 0.3  # Found something, but it's wrong
    else:
        return 0.0  # Missing entirely
```
### 3) Track "Why" (The Reason Layer)
For the next step, track why something scored low. This will be essential for debugging, gathering training data, and improving the Resolver V1 later.

**Example structured output:**
```json
{
  "field": "skills",
  "score": 0.47,
  "notes": "missed 3 expected skills, 2 fuzzy matches"
}
```
### 💡 Why this matters
* **Nuance:** We get credit for "near misses" rather than flat zeros.
* **Debuggability:** We can identify if a specific field (like Location) is failing across the entire dataset.
* **Metric Integrity:** This provides the "Ground Truth" needed to safely transition from Regex to AI-augmented parsing.

### ✅ How to tell it’s done
* The benchmark script output now includes a per-field breakdown.
* The total benchmark score is a weighted average of these field scores.
* Scores below **1.0** include a `notes` or `reason` entry in the logs.

**🏷️ Labels:** `enhancement`, `benchmarks`, `math-heavy`

**Comment by Kevin-TCUS** [comment]
Closing as superseded by the current benchmark harness direction.

This issue was opened when the benchmark concept was still early and proposed fuzzy field-level scoring across name, email, location, and skills.

The v0.3 benchmark direction has since become more specific and measurable:

- focus on skills and location first
- report TP / FP / FN examples
- compute precision / recall / F1
- preserve raw scoring
- add resolver-aware skills scoring separately through #233

That approach gives clearer parser/resolver signal than a single weighted fuzzy score. Future field-specific scoring can be reintroduced through narrower follow-up issues if needed.
